astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,302 @@
+          'venus': [(0, 2), (2, 299)],
+          'earth-moon-barycenter': [(0, 3)],
+          'earth':  [(0, 3), (3, 399)],
+          'moon': [(0, 3), (3, 301)],
+          'mars': [(0, 4)],
+          'jupiter': [(0, 5)],
+          'saturn': [(0, 6)],
+          'uranus': [(0, 7)],
+          'neptune': [(0, 8)],
+          'pluto': [(0, 9)]}
+
+
+def _download_spk_file(show_progress=True):
+    """
+    Get the Satellite Planet Kernel (SPK) file `de430.bsp` from NASA JPL.
+
+    Download the file from the JPL webpage once and subsequently access a
+    cached copy. This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+
+    .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
+    """
+    de430_url = ('http://naif.jpl.nasa.gov/pub/naif/'
+                 'generic_kernels/spk/planets/de430.bsp')
+    return download_file(de430_url, cache=True, show_progress=show_progress)
+
+
+def _get_kernel(*args, **kwargs):
+    """
+    Try importing jplephem, download/retrieve from cache the Satellite Planet
+    Kernel.
+    """
+    global KERNEL
+
+    try:
+        from jplephem.spk import SPK
+    except ImportError:
+        raise ImportError("Solar system ephemeris calculations depend on"
+                          "the jplephem package")
+
+    if KERNEL is None:
+        KERNEL = SPK.open(_download_spk_file())
+    return KERNEL
+
+
+def get_barycentric_body_position(time, body_key):
+    """
+    Calculate the barycentric position of solar system body ``body_key`` in cartesian coordinates.
+
+    Uses ``jplephem`` with the DE430 kernel.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body_key : int, str
+        The solar system body to calculate. Can be a string, e.g. 'moon',
+        or a valid integer index.
+
+        See help for ``get_body`` for a listing of valid integer indices.
+
+    Returns
+    -------
+    cartesian_position : `~astropy.coordinates.CartesianRepresentation`
+        Barycentric (ICRS) position of the body in cartesian coordinates
+
+    Notes
+    -----
+
+    """
+    # if we have an int as a body index, then create chain to that body
+    if isinstance(body_key, int):
+        # chain consists of single step, from 0 (solar system barycenter) to body_key
+        chain = [(0, body_key)]
+    elif isinstance(body_key, six.string_types):
+        try:
+            chain = BODIES[body_key]
+        except KeyError:
+            raise ValueError("Unknown body index '" + body_key + """', valid entries are
+                              contained within solar_system.BODIES""")
+    else:
+        raise ValueError("'body_key' must be a string or integer")
+
+    kernel = _get_kernel()
+
+    # are all the kernel pairs in this kernel?
+    valid_chain = np.all([c in kernel.pairs.keys() for c in chain])
+    if not valid_chain:
+        raise ValueError("Postion of this body cannot be calculated using jpl kernel")
```

Comment:
Remove it now we don't use `int` any more?  Or leave it in case someone uses a different kernel? But then perhaps we should also check for out-of-range times? My sense would be to remove this check.